File: R/mensagem.R
mensagem_progresso_endbr <- function(msg) {
  rlib_verboso <- (getOption("rlib_message_verbosity", "quiet") == "verbose")
  pacote_verboso <- (getOption("enderecobr.verbose", "quiet") == "verbose")

  if (rlib_verboso || pacote_verboso) {
    cli::cli_progress_step(msg, .envir = parent.frame())
  }
}

File: python-package/src/enderecobr/mensagem.py
# --- Mensagem de progresso ------------------------------------------------

def mensagem_progresso_endbr(msg: str) -> None:
    """
    Se getOption("rlib_message_verbosity") ou getOption("enderecobr.verbose")
    estiverem em "verbose", mostra msg na tela (equivalente ao cli_progress_step()).
    """
    rlib_verbose = os.getenv("RLIB_MESSAGE_VERBOSITY", "quiet") == "verbose"
    pkg_verbose = os.getenv("ENDERECOBR_VERBOSE",      "quiet") == "verbose"

    if rlib_verbose or pkg_verbose:
        # se tiver click instalado, usar click.echo() para cores e flush automático
        try:
            import click
            click.echo(msg)
        except ImportError:
            print(msg)

